### PR TITLE
feat: TE7-T1022: Implement Create User Screen Redux

### PR DESCRIPTION
**Pull request by jigar-s-simform**
Implement create user screen redux logic

### CODE REVIEW

File: app/constants/Strings.ts [hunk 1]
```diff
@@ -57,5 +57,10 @@ export default {
   gallery: 'Gallery',
   camera: 'Camera',
   avatar: 'Avatar',
-  featureUnavaible:'This Feature Unavailble/Unsupported on your device!'
+  featureUnavaible: 'This Feature Unavailble/Unsupported on your device!',
```
Comment (amitkajwani-simformsolutions) [inline app/constants/Strings.ts]: Check for a typo mistake.

File: app/modules/create-user/PictureOptionsModal.tsx [hunk 2]
```diff
@@ -8,32 +8,48 @@ import styles from './CreateUserScreenStyles';
...
+const PictureOptionsModal = ({
+  modalShown,
+  setModalShown,
+  handleCameraSelect,
+  handleGallerySelect,
+}: ModalPropsType): JSX.Element => {
+
   const handleModalVisibility = (): void => {
     setModalShown(!modalShown);
   };
```
Comment (amitkajwani-simformsolutions) [inline app/modules/create-user/PictureOptionsModal.tsx:24]: Add one blank line

File: app/modules/create-user/useCreate.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  const galleryPermisionStatus: PermissionStatus =
+    usePermissionStatus(galleryPermision);
+
+  const [modalShown, setModalShown] = useState<boolean>(false);
+  const formRefs: React.MutableRefObject<any[]> = useInitializeRefs(4);
+
+  const openDeviceCamera = async (): Promise<void> => {
+    try {
+      const imageCaptureResponse = await launchCamera({
+        mediaType: 'photo',
```
Comment (amitkajwani-simformsolutions) [inline app/modules/create-user/useCreate.ts]: For mediaType, we can create an immutable object in constant and use it from there. [sort of enum]

File: app/modules/create-user/useCreate.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        cameraType: 'back',
```
Comment (amitkajwani-simformsolutions) [inline app/modules/create-user/useCreate.ts]: For cameraType, we can create an immutable object in constant and use it from there. [sort of enum]

File: ios/Ts_Assignment/Info.plist [hunk 5]
```diff
@@ -4,6 +4,8 @@
 <dict>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
+	<key>NSAppleMusicUsageDescription</key>
```
Comment (amitkajwani-simformsolutions) [inline ios/Ts_Assignment/Info.plist]: why apple music?

File: app/utils/requestPermission.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { Permission, request } from "react-native-permissions";
+
+export const requestPermission = async (permission : Permission) => {
```
Comment (amitkajwani-simformsolutions) [inline app/utils/requestPermission.ts]: remove space before :

File: app/modules/create-user/useCreate.ts (hunk 4 above)
Comment (amitkajwani-simformsolutions) [reply app/modules/create-user/useCreate.ts]: Check at all places

File: app/modules/create-user/useCreate.ts (hunk 3 above)
Comment (amitkajwani-simformsolutions) [reply app/modules/create-user/useCreate.ts]: Check at all places

**Review by amitkajwani-simformsolutions — CHANGES_REQUESTED** [review]

File: ios/Ts_Assignment/Info.plist (hunk 5 above)
Comment (jigar-s-simform) [reply ios/Ts_Assignment/Info.plist]: because it was giving an error and explicitly told me to add this

File: app/constants/Media.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+export const MediaTypes = Object.freeze({
+  photo: 'photo',
+  video: 'video',
+  mixed: 'mixed',
+});
+
+export const CameraTypes = Object.freeze({
+  back: 'back',
+  front:'front'
```
Comment (amitkajwani-simformsolutions) [inline app/constants/Media.ts]: space after :

File: app/modules/profile/ProfileScreen.tsx [hunk 8]
```diff
@@ -23,7 +24,7 @@ const ProfileScreen:FC = () => {
       <View style={styles.top}>
         <View style={styles.profileImgContainer}>
           <Image
-            source={{uri: userDetails?.avatar}}
+            source={typeof userDetails?.avatar === 'string' ?{uri: userDetails?.avatar} : Images.defaultImg}
```
Comment (amitkajwani-simformsolutions) [inline app/modules/profile/ProfileScreen.tsx]: space after ?

File: ios/Ts_Assignment/Info.plist (hunk 5 above)
Comment (amitkajwani-simformsolutions) [reply ios/Ts_Assignment/Info.plist]: Where it was asking, I mean which library?

**Review by amitkajwani-simformsolutions — CHANGES_REQUESTED** [review]

File: ios/Ts_Assignment/Info.plist (hunk 5 above)
Comment (jigar-s-simform) [reply ios/Ts_Assignment/Info.plist]: I was trying to install react-native-permissions. I cannot explain why! but it was not working without that! you can experience the same by trying to install react-native-permissions

Comment (amitkajwani-simformsolutions) [reply ios/Ts_Assignment/Info.plist]: Please share the error that exactly you are facing.

**Review by amitkajwani-simformsolutions — APPROVED** [review]

**Pull request merged**